lib/spec_file_generator.js

```diff
@@ -1,163 +1,116 @@
-    '\\$VERSION': npm_module.version,
-    '\\$RELEASE': release,
-    '\\$LICENSEFILE': npm_module.license_file.length !== 0 ? '%license ' + npm_module.license_file : '',
-    '\\$LICENSETYPE': npm_module.license,
-    '\\$SUMMARY': npm_module.summary,
-    '\\$PROJECTURL': npm_module.project_url,
-    '\\$DESCRIPTION': npm_module.description,
-    '\\$BINSNIPPET': symlinkBinaries(npm_module),
-    '\\$FILESBINSNIPPET': listBinaryFiles(npm_module.binaries),
-    '\\$COPYFILES': copyFiles(npm_module),
-    '\\$DOCFILES': listDocFiles(npm_module.doc_files),
-    '\\$DATE': Date.today().toString("ddd MMM d yyyy"),
+    NAME: npm_module.name,
+    VERSION: npm_module.version,
+    RELEASE: release,
+    LICENSEFILES: getLicenceFiles(files),
+    LICENSETYPE: npm_module.license,
+    SUMMARY: getSummary(npm_module.description),
+    PROJECTURL: npm_module.homepage,
+    DESCRIPTION: npm_module.description,
+    BINARIES: npm_module.bin,
+    COPYFILES: getCopyFiles(files),
+    DOCFILES: getDocFiles(files),
+    DATE: Date.today().toString("ddd MMM d yyyy")
   }
 
-  if (npm_module.bundle) {
-    replacements['\\$PROVIDES'] = dependenciesToProvides(dependencies);
-    replacements['\\$SOURCES'] = dependenciesToSources(dependencies);
-    replacements['\\$BUILD'] = npmInstallCache();
-    replacements['\\$SETUP'] = sourcesToCache(dependencies);
-    replacements['\\$SYMLINK'] = '';
+  if (dependencies.length !== 0) {
+    replacements['DEPENDENCIES'] = [];
+    replacements['PROVIDES'] = dependenciesToProvides(dependencies);
+    replacements['SOURCES'] = dependenciesToSources(dependencies);
   } else {
-    replacements['\\$PROVIDES'] = '%{?nodejs_find_provides_and_requires}';
-    replacements['\\$SOURCES'] = 'Source0: ' + helpers.npmUrl('%{npm_name}', '%{version}');
-    replacements['\\$BUILD'] = '';
-    replacements['\\$SETUP'] = '%setup -q -n package';
-    replacements['\\$SYMLINK'] = '%nodejs_symlink_deps';
+    replacements['DEPENDENCIES'] = dependenciesToRequires(npm_module['dependencies']);
+    replacements['PROVIDES'] = [];
+    replacements['SOURCES'] = ['Source0: ' + helpers.npmUrl('%{npm_name}', '%{version}')];
   }
 
   var spec_file = fs.readFileSync(template, { encoding: 'utf8' });
-
-  for (var key in replacements) {
-    var replacement = replacements[key];
-    if (replacement === undefined) {
-      replacement = 'FIXME';
-    }
-    spec_file = replaceAttribute(spec_file, key, replacement);
-  }
-
-  return spec_file;
-}
-
-function npmInstallCache() {
-  return 'npm install --cache-min Infinity --cache %{npm_cache_dir} --no-optional --global-style true %{npm_name}@%{version}';
-}
-
-function sourcesToCache(deps) {
-  return 'mkdir -p %{npm_cache_dir}\n\
-for tgz in %{sources}; do\n\
-  echo $tgz | grep -q registry.npmjs.org || npm cache add --cache %{npm_cache_dir} $tgz\n\
-done\n\
-\n\
-%setup -T -q -a ' + deps.length + ' -D -n %{npm_cache_dir}';
+  var template = Handlebars.compile(spec_file);
```

Comment:
`template` is is already defined in the function signature (it's used in the line above this even), could you rename it?